server/config.py

```diff
@@ -0,0 +1,73 @@
+"""Module providing primary input and output configuration paths."""
+import dataclasses
+import os
+
+INDEX_DIR_ENV = 'INDEX_DIR'
+TRANSCRIPTS_DIR_ENV = 'TRANSCRIPTS_DIR'
+UPLOAD_DIR_ENV = 'UPLOAD_DIR'
+RECORDINGS_DIR_ENV = 'RECORDINGS_DIR'
+
+
+@dataclasses.dataclass
+class APIConfig:
+    """Class representing third-party API keys and other settings."""
+    daily_api_key: str = None
+    daily_api_url: str = None
+    deepgram_api_key: str = None
+    deepgram_model_name: str = None
+
+
+def get_third_party_config() -> APIConfig:
+    """Returns third-party configuration"""
+    return APIConfig(
+        os.getenv("DAILY_API_KEY"),
+        os.getenv("DAILY_API_URL"),
+        os.getenv("DEEPGRAM_API_KEY"),
+        os.getenv("DEEPGRAM_MODEL_NAME")
+    )
+
+
+def ensure_dirs():
+    """Creates required file directories if they do not already exist."""
+    ensure_dir(TRANSCRIPTS_DIR_ENV)
+    ensure_dir(INDEX_DIR_ENV)
+    ensure_dir(UPLOAD_DIR_ENV)
+    ensure_dir(RECORDINGS_DIR_ENV)
+
+
+def ensure_dir(env_name: str):
+    """Creates directory based on env variable,
+    if said directory does not already exist."""
+    directory = os.getenv(env_name)
+    if not directory:
+        directory = env_name
+        os.environ[env_name] = directory
+
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def get_transcripts_dir_path() -> str:
+    """Returns transcript directory path."""
+    return os.path.abspath(os.getenv(TRANSCRIPTS_DIR_ENV))
```

Comment:
I believe all these should also be part of the `APIConfig`. Because, again, I would not be able to build a valid `APIConfig` without environment variables.

So, this could be:

```
def get_transcripts_dir_path() -> str:
    """Returns transcript directory path."""
    return os.path.abspath(self.transcripts_dir)
```

Reply:
These are not part of any third-party API though (which was the thinking behind the name `APIConfig`, and the only one that consumes them is this specific file. I think including them in what essentially turns into a global object errs on exposing the side of too many variables to things that don't need them (additionally introducing a risk of accidental manipulation). Like, if the local directory path variables are only ever needed in this one file, I sort of prefer not to expose them and pass them around further.

Reply:
Actually... I think you're right. I will rename this class, make a single Config class out of this file (more appropriate naming), and possibly make use of a read-only property approach or just the underscore prefix convention.

Reply:
Yeah, doesn't need to be only API keys, could be anything the application needs as configuration. :raised_hands:

Reply:
Alright, I submitted a new config class. Overview:

* The primary consumers of the `Config` instance are `index.py` server routes and the vector store. All other modules receive the variables they need directly vs the entire `Config` object (for isolation purposes and because they just don't need to know).
* All env variables are now retrieved only once, when initializing the config class, if no other values for each property are provided to the constructor.
* Default fallbacks remain the same for things like local directories.
* Using underscore convention for intended-private config variables and property accessors to retrieve them.

Hope that makes sense!

Reply:
Nice, thanks!